anima/ui/user_dialog.py

```diff
@@ -0,0 +1,167 @@
+                                   "font: 18pt;")        
+        self.dialog_label.setText("Create User")
+        
+        self.line = QtWidgets.QFrame()
+        self.line.setFrameShape(QtWidgets.QFrame.HLine)
+        self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
+        
+        self.name_line_edit=QtWidgets.QLineEdit()
+        self.name_line_edit.setPlaceholderText("Enter Name")
+        
+        self.login_line_edit=QtWidgets.QLineEdit()
+        self.login_line_edit.setPlaceholderText("stalker")
+        
+        self.email_line_edit=QtWidgets.QLineEdit()
+        self.email_line_edit.setPlaceholderText("stalker@gmail.com")
+        
+        self.password_line_edit=QtWidgets.QLineEdit()
+        self.password_line_edit.setPlaceholderText("******")
+        
+        self.ok_button=QtWidgets.QPushButton("OK")
+        self.cancel_button=QtWidgets.QPushButton("Cancel")
+               
+       
+        
+    def create_layouts(self):
+
+        form_layout=QtWidgets.QFormLayout()
+        form_layout.addRow("Name", self.name_line_edit)
+        form_layout.addRow("Login", self.login_line_edit)
+        form_layout.addRow("Email", self.email_line_edit)
+        form_layout.addRow("Password", self.password_line_edit)
+        
+        button_layout=QtWidgets.QHBoxLayout()
+        button_layout.addStretch()
+        button_layout.addWidget(self.ok_button)
+        button_layout.addWidget(self.cancel_button)
+        
+        label_layout=QtWidgets.QVBoxLayout()
+        label_layout.addWidget(self.dialog_label)
+        label_layout.addWidget(self.line)
+        label_layout.addStretch()
+        
+        main_layout=QtWidgets.QVBoxLayout(self)
+        main_layout.addLayout(label_layout)
+        main_layout.addLayout(form_layout)
+        main_layout.addLayout(button_layout)
+        
+
+        self.resize(328,184)
+        
+        
+    def create_connections(self):
+        
+        self.ok_button.clicked.connect(self.check_userPass)
+        self.cancel_button.clicked.connect(self.close)
+        
+        
+    def check_userPass(self):
+        
+        self.user_name=self.name_line_edit.text()
+        self.user_login=self.login_line_edit.text()
+        self.user_email=self.email_line_edit.text()
+        self.user_password=self.password_line_edit.text()
+        
+        if len(self.user_name)<2 or len(self.user_login)<2 or len(self.user_email)<2 or len(self.user_password)<2 :
+            msg = QtWidgets.QMessageBox()
+            msg.setIcon(QtWidgets.QMessageBox.Information)
+
+            msg.setText("Please, fill out user information completely!")
+            msg.setWindowTitle("Information Message")
+            msg.setStandardButtons(QtWidgets.QMessageBox.Ok) 
+            msg.exec_() 
+            
+        else:
+            try:
+                self.create_user()
+            except:
+                pass
+                
+            self.close()
```

Comment:
There is something wrong with this code, it will exit prematurely.

Think that the ``self.create_user()`` on line 134 throw an exception. Because, basically you're skipping all of the errors in between line 133 and 136 (hence the try->except->pass) and then you're closing the dialog prematurely and showing a message that a user is created successfully.

The code should have been like this may be:

```python
try:
    self.create_user()
except BaseException as e:
    QtWidgets.QMessageBox.critical(
        self,
        "Error",
        str(e) # I don't know if this is still the correct way
    )          # of getting the string representation of an exception

else:
    # now we can give the information message
    QtWidgets.QMessageBox.information(
        self,
        "Success",
        "User '{0}' successfully created!".format(self.user_login)
    )
    # then we can close this dialog
    self.close() # or may be self.accept()
```